### PR TITLE
Making mark post read fields optional.

### PR DESCRIPTION
**Pull request by dessalines**
Ran into this issue with lemmy-ui : it required a `post_id` and `post_ids`, neither optional.

I'd rather not do it this way, and just remove the old one now, but this lets them both stay for the time being.

### CODE REVIEW

File: crates/api_common/src/post.rs [hunk 1]
```diff
@@ -135,14 +135,15 @@ pub struct RemovePost {
   pub reason: Option<String>,
 }
 
+#[skip_serializing_none]
```
Comment (Nutomic) [inline crates/api_common/src/post.rs:138]: Seems unnecessary for something that is only received, not sent.

File: crates/api/src/post/mark_read.rs [hunk 2]
```diff
@@ -11,14 +11,25 @@ pub async fn mark_post_as_read(
...
+    post_ids.extend(post_ids_.iter().cloned());
+  }
+
+  if let Some(post_id) = data.post_id {
+    post_ids.insert(post_id);
+  }
+
+  if post_ids.is_empty() {
+    Err(LemmyErrorType::CouldntMarkPostAsRead)?;
+  }
```
Comment (Nutomic) [inline crates/api/src/post/mark_read.rs:25]: Imo we can leave this check out.

File: crates/api_common/src/post.rs (hunk 1 above)
Comment (dessalines) [reply crates/api_common/src/post.rs:138]: Its necessary for ts-rs.

File: crates/api_common/src/post.rs [hunk 3]
```diff
@@ -135,14 +135,15 @@ pub struct RemovePost {
...
 #[derive(Debug, Serialize, Deserialize, Clone, Default)]
 #[cfg_attr(feature = "full", derive(TS))]
 #[cfg_attr(feature = "full", ts(export))]
 /// Mark a post as read.
 pub struct MarkPostAsRead {
   /// TODO: deprecated, send `post_ids` instead
-  pub post_id: PostId,
-  pub post_ids: Vec<PostId>,
+  pub post_id: Option<PostId>,
+  pub post_ids: Option<Vec<PostId>>,
```
[the remark this answers is not in the document]
Comment (dessalines) [reply crates/api_common/src/post.rs:146]: I tried, but it didn't make it any simpler. That also isn't on any other field in api_common.

[the remark this answers is not in the document]
Comment (Nutomic) [reply crates/api_common/src/post.rs]: Its a core serde feature and widely used in federation code, no reason why it cant be used here. Pushed a commit to your branch to add this, and its definitely simpler.

**Review by Nutomic — APPROVED** [review]

File: crates/api_common/src/post.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (Nutomic) [reply crates/api_common/src/post.rs]: `#[serde(default)]` means that the vec is optional, see the [docs](https://serde.rs/field-attrs.html#default). Or is this a problem with ts-rs not understanding serde attributes?

[the remark this answers is not in the document]
Comment (dessalines) [reply crates/api_common/src/post.rs]: Didn't work, it needs to be optional.

Screenshot_2023-10-19-11-35-05-047_com termux

**Pull request merged**